Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -909,7 +910,18 @@ void ParseAttachPointLine(RWMol *mol, const std::string &text, unsigned int line
                << " not found";
         throw FileParseException(errout.str());
       } else {
-        atom->setProp(common_properties::molAttachPoint,val);
+        if (val == -1) {
+          // this is 3 in v3k mol blocks, so use that:
+          val = 3;
```

Comment:
Sorry, maybe I didn't explain myself correctly. It is actually the opposite: The atom block uses "-1",  the "M  APO" line uses 3, so, if we want to use the same criteria we use in the atom block this should be:

```suggestion
        if (val == 3) {
          // this is -1 in v3k mol blocks, so use that:
          val = -1;
```

And move this check after we made sure `val` has an allowed value.